src/state/swap/hooks.ts

```diff
@@ -314,15 +322,22 @@ export function useDerivedSwapInfo(platformOverride?: RoutablePlatform): UseDeri
     loading,
   }
 
-  function getTradesPromise(
+  async function getTradesPromise(
     parsedAmount: CurrencyAmount,
     inputCurrency: Currency,
     outputCurrency: Currency,
     commonParams: { maximumSlippage: Percent; receiver: string; user: string },
     ecoRouterSourceOptionsParams: { uniswapV2: { useMultihops: boolean } },
-    staticJsonRpcProvider: StaticJsonRpcProvider | undefined
+    staticJsonRpcProvider: StaticJsonRpcProvider | undefined,
+    signal: AbortSignal
   ): Promise<EcoRouterResults> {
-    return isExactIn
+    const abortPromise = new Promise<EcoRouterResults>((resolve, reject) => {
+      signal.onabort = () => {
+        reject(new DOMException('Aborted', 'AbortError'))
+      }
+    })
```

Comment:
Do we need to log the Aborted message?

Reply:
We can also call `resolve()` and check if the are no trades. But I think it's better to throw an error and check for the message, because the promise wasn't successful.